test/e2e/pd.go

```diff
@@ -0,0 +1,217 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("PD", func() {
+	var (
+		c         *client.Client
+		host0Name string
+		host1Name string
+	)
+
+	BeforeEach(func() {
+		var err error
+		c, err = loadClient()
+		Expect(err).NotTo(HaveOccurred())
+
+		nodes, err := c.Nodes().List()
+		expectNoError(err, "Failed to list nodes for e2e cluster.")
+		Expect(len(nodes.Items) >= 2)
+
+		host0Name = nodes.Items[0].ObjectMeta.Name
+		host1Name = nodes.Items[1].ObjectMeta.Name
+	})
+
+	It("should schedule a pod w/ a RW PD, remove it, then schedule it on another host", func() {
+		podClient := c.Pods(api.NamespaceDefault)
+		diskName := fmt.Sprintf("e2e-%d", time.Now().Unix())
+
+		host0Pod := testPDPod(diskName, host0Name, false)
+		host1Pod := testPDPod(diskName, host1Name, false)
+
+		By("creating PD")
+		err := createPD(diskName, testContext.gceConfig.Zone)
+		expectNoError(err, fmt.Sprintf("Error creating PD: %v", err))
+
+		defer func() {
+			By("cleaning up PD-RW test environment")
+			defer GinkgoRecover()
```

Comment:
These aren't needed unless you're on a goroutine (you're probably looking at some test we haven't cleaned up, sorry)